### ISSUE TITLE
cmd/go: get ./foo doesn't download foo

### ISSUE THREAD

**Issue by rsc** [golang/go#19769]
```
$ rm -rf /tmp/x
$ mkdir -p /tmp/x/src/rsc.io
$ cd /tmp/x/src/rsc.io
$ GOPATH=/tmp/x go1.5 get ./pprof_mac_fix
$ rm -rf pprof_mac_fix
$ GOPATH=/tmp/x go1.7 get ./pprof_mac_fix
$ rm -rf pprof_mac_fix
$ GOPATH=/tmp/x go1.8 get ./pprof_mac_fix
$ rm -rf pprof_mac_fix
$ GOPATH=/tmp/x go get ./pprof_mac_fix
can't load package: package ./pprof_mac_fix: cannot find package "./pprof_mac_fix" in:
	/tmp/x/src/rsc.io/pprof_mac_fix
$ ls
$ 
```

A recurrence of #9767 except in that case the download happened and then go get failed to build the downloaded code. Now it's not downloading the code at all.

There is a test for this in cmd/go already but it doesn't run on builders because it needs an external network.

**Comment by rsc** [comment]
This bisects to CL 33158, which also caused other problems. I wonder if we should roll back that CL instead. 

/cc @shurcooL

**Comment by shurcooL** [comment]
I didn't see this issue before. At first glance, it looks like a duplicate of issue #20175, which I was notified about and sent a fix for. I'm still waiting on your review of that CL. https://go-review.googlesource.com/c/42350/ Can you look at it first?

**Comment by shurcooL** [comment]
I've confirmed, this issue is the same as #20175, and my fix in [CL 42350](https://golang.org/cls/42350) fixes it too.

This is `cmd/go` on tip:

```
$ go test -v -run=TestGoGetDotSlashDownload cmd/go
=== RUN   TestGoGetDotSlashDownload
--- FAIL: TestGoGetDotSlashDownload (0.05s)
	go_test.go:310: running testgo [get ./pprof_mac_fix]
	go_test.go:323: standard error:
	go_test.go:324: can't load package: package ./pprof_mac_fix: cannot find package "./pprof_mac_fix" in:
			/var/folders/tw/kgz4v2kn4n7d7ryg5k_z3dk40000gn/T/gotest134427085/src/rsc.io/pprof_mac_fix
		
	go_test.go:333: go [get ./pprof_mac_fix] failed unexpectedly: exit status 1
FAIL
exit status 1
FAIL	cmd/go	1.701s
```

And this is with CL 42350:

```
$ go test -v -run=TestGoGetDotSlashDownload cmd/go
=== RUN   TestGoGetDotSlashDownload
--- PASS: TestGoGetDotSlashDownload (5.02s)
	go_test.go:269: running testgo [get ./pprof_mac_fix]
PASS
ok  	cmd/go	15.612s
```

Should we close one of the two issues as duplicate? If so, which one; this issue was earlier, but the other has more information posted in it. Or let the CL resolve both?

**Comment by rsc** [comment]
Thanks for the fix, and sorry about the delayed review. I hadn't pinged you on this because I didn't bisect it until this morning.

**Comment by shurcooL** [comment]
@rsc No problem, I'm happy to follow up and resolve any issues my work might (unintentionally) cause. Thanks!
